Kubernetes Installer v2023.03.28-0 release notes (#1043)
replicatedhq/replicated-docs: docs/release-notes/rn-kubernetes-installer.md

# Kubernetes Installer Release Notes

## v2023.03.28-0

Released on March 28, 2023

### New Features {#new-features-v2023-03-28-0}
* Adds [MinIO add-on](https://kurl.sh/docs/add-ons/minio) version RELEASE.2023-03-20T20-16-18Z.
* Adds [Rook add-on](https://kurl.sh/docs/add-ons/rook) verison 1.11.2.
* Adds [Weave add-on](https://kurl.sh/docs/add-ons/weave) version.
* Adds [Flannel add-on](https://kurl.sh/docs/add-ons/flannel) version 0.21.4.
* Adds [Metrics Server add-on](https://kurl.sh/docs/add-ons/metrics-server) version 0.6.3.

### Improvements {#improvements-v2023-03-28-0}

> * Adds preflight checks to prevent installations which requires an Object Store.

Reviewer: ```suggestion
* Adds preflight checks to prevent installations which requires an Object Store be done without it. Note that installation which are done with Kots and with the option disableS3 requires to have an object store set. Also, see that is unsupported migrate from Rook to OpenEBS using the Registry add-on without set MinIO.
```

> * Adds preflight checks to prevent installations which requires an Object Store.

Reviewer 2: ```suggestion
* Adds preflight checks to prevent installations without the kotsadm.disableS3 option set to continue without an Object Store.
```

> * Adds preflight checks to prevent installations without the kotsadm.disableS3 option set to continue without an Object Store.

Reviewer 2: ```suggestion
* Adds preflight checks to prevent installations without the kotsadm.disableS3 option set from continuing without an Object Store.
```

> * Adds preflight checks to prevent installations without the kotsadm.disableS3 option set from continuing without an Object Store.

Reviewer 2: ```suggestion
* Adds preflight checks to prevent installations without the `kotsadm.disableS3` option set to `true` from continuing without an Object Store.
```

* Adds preflight checks to prevent migrating from Rook to OpenEBS without MinIO when the Registry add-on is included in the spec.

> * The optional flag `force-reapply-addons` has been removed and it is now the default behavior to re-apply all add-ons regardless of whether or not they are changing.

Reviewer: Should we not say that `flag `force-reapply-addons` is deprecated?

> 

Reviewer: Should we not add here that now the status of the migrations from Rook are tracked in the ConfigMap  `kurl-migration-from-rook`  placed on the kurl namespace? So that, users and CRE are able to be aware of that and indeed can check it if/when face issues. WDYT?

Reviewer 2: i dont see what value that delivers to the end customer. this is not a changelog, it is release notes

### Bug Fixes {#bug-fixes-v2023-03-28-0}

> * Fixes an issue when upgrading from Kubernetes 1.20.x to 1.21.x that causes the script to fail with error "connection refused".

Reviewer: @emosbaugh I do not think that we should here say the specific k8s versions because we fixed the issue which we cannot ensure that only occurs in this case. IMO would be better say the error faced and when so that I am as user or CRE I can see in the release notes that scenario is fixed. 
```suggestion
* Fixes an issue when upgrading from Kubernetes releases that causes the script to fail with error "connection refused" and the message "couldn't retrieve DNS addon deployments"
```

# Kubernetes Installer Release Notes

## v2023.03.28-0

Released on March 28, 2023

### New Features {#new-features-v2023-03-28-0}

> * Adds host preflights checks to prevent new installs which requires an Object Store be done without it.

Reviewer: ```suggestion
* Adds preflights checks to prevent installs which requires an Object Store set
```

* Adds [MinIO add-on](https://kurl.sh/docs/add-ons/minio) version RELEASE.2023-03-20T20-16-18Z.

> * Support has been added for Red Hat Enterprise Linux 9 and Rocky Linux 9.

Reviewer: ```suggestion
```

> * Adds preflight checks to prevent migrating from Rook to OpenEBS with Registry without MinIO.

Reviewer: ```suggestion
* Adds preflight checks to prevent migrating from Rook to OpenEBS using Registry without have MinIO set.
```

> * Adds [Containerd add-on](https://kurl.sh/docs/add-ons/containerd) version 1.6.19.

Reviewer: ```suggestion
* Adds migration from Rook status to a ConfigMap placed on the kurl namespace named as `kurl-migration-from-rook` 
```

> * The kURL installer will now force reapply addons by default deprecating the usage of the `force-reapply-addons` options.

Reviewer: ```suggestion
* Adds usage of reapply add-on(s) option forced by default. 
* Deprecates the usage of the optional flag `force-reapply-addons`.
```

* Adds [Rook add-on](https://kurl.sh/docs/add-ons/rook) verison 1.11.2.
* Adds [Weave add-on](https://kurl.sh/docs/add-ons/weave) version.
* Adds [Flannel add-on](https://kurl.sh/docs/add-ons/flannel) version 0.21.4.
* Adds [Metrics Server add-on](https://kurl.sh/docs/add-ons/metrics-server) version 0.6.3.

### Bug Fixes {#bug-fixes-v2023-03-28-0}

> * Fixes `kubeadm init` addons phase failure when upgrading Kubernetes.

Reviewer: ```suggestion
* Fixes error faced to upgrade where are NotReady control-planes in the cluster` 
```

> * Adds status of migration from rook into configmap.

Reviewer: ```suggestion
```

* Fixes an issue that could cause the install script to exit with an error when running preflights if the kubectl is installed but not kubernetes or the cluster is down.
* Fixes migration from Rook issues where Rook should be removed and it is not by automating troubleshooting steps when timeouts are faced and ensuring that Ceph Block Pool is removed. This fixes issues constantly faced when migrating from Rook versions 1.4.9 and upper.

> * Adds check to not allow perform more than once successfully the object store migration.

Reviewer: ```suggestion
* Fixes scenario where the object store in the storage migrations could be migrated more than once.
```
